openmrs-19/src/main/java/org/motechproject/openmrs19/resource/impl/PatientResourceImpl.java

```diff
@@ -83,6 +87,27 @@ public String getMotechPatientIdentifierUuid() throws HttpException {
     }
 
     @Override
+    public String getPatientIdentifierName(String uuid) throws HttpException {
+        String identifierName = identifierUuidByName.get(uuid);
+
+        if (identifierName == null) {
+            PatientIdentifierListResult result = getAllPatientIdentifierTypes();
+            for (IdentifierType type : result.getResults()) {
+                if (StringUtils.equals(uuid, type.getUuid())) {
+                    if (openmrsInstance.getPatientIdentifierTypesNames().contains(type.getName())) {
+                        identifierName = type.getName();
```

Comment:
Is it supposed to return identifier name, based on the value of that identifier? What if I have the same value for various identifiers? Eg. MotechID=1;  RandomID=1;  OneMoreID=1?